### DISCUSSION TITLE
How to get access to hit/miss kprobe stat while using pmuKprobe

### DISCUSSION THREAD

**Discussion by nloginov** [cilium/ebpf#880] [Q&A] [answered]
Hello, community of cilium/ebpf!

Previously I used https://github.com/iovisor/gobpf library, and that library created kprobe through `tracefs`, so that I was able to monitor statistic about total count of hit and miss kprobes through  `/sys/kernel/debug/tracing/kprobe_profile` file.

I found out that by default cillium/ebpf is creating kprobes using PMU, and it falls back to the tracefs only if PMU method is not supported:
https://github.com/cilium/ebpf/blob/aaa9af24fdd8e397be823526928782cc93d59188/link/kprobe.go#L184-L199

I assume that it is a more modern and proper way of creating kprobe, but could you tell me how I can access to the hit/miss statistic using this approach? 

Let's say my ebpf program crashed during execution (e.g. referenced invalid memory address): with tracefs method this failed execution will fall under "miss" counter in  `/sys/kernel/debug/tracing/kprobe_profile` file, and I will be able to monitor that. With PMU method no such file is created and I have no idea where I can get this information or is it possible at all, could you help me? Thanks in advance!

**Comment by lmb** [comment, marked as answer]
That's a fair question. As far as I know the kernel currently doesn't expose that kind of information, see https://github.com/cilium/ebpf/pull/755#issuecomment-1209099541

(Thanks to  @mmat11 for digging up the necessary information).

**Reply by mmat11** [reply]
From this message https://github.com/cilium/ebpf/pull/755#issuecomment-1209099541 I understand it's not currently possible
